libs/blocks/commerce/commerce.js

```diff
@@ -0,0 +1,139 @@
+};
+
+export const decorateOfferDetails = async (el, of, searchParams) => {
+  function formatOfferDetailKeys(str) {
+    const details = str.split(/(?=[A-Z])/);
+    const allCapsDetail = details.map((detail) => detail.toUpperCase());
+    const result = allCapsDetail.join(' ');
+    return result;
+  }
+  const offerDetailsList = document.createElement('ul');
+  offerDetailsList.className = 'offer-details';
+  const offer = formatOfferDetails(of);
+  Object.entries(offer).forEach(([key, value]) => {
+    const offerData = document.createElement('li');
+    const offerKey = document.createElement('span');
+    const offerValue = document.createElement('span');
+    offerData.classList.add('offer-detail');
+    offerKey.classList.add('offer-key');
+    offerValue.classList.add('offer-value');
+    offerKey.textContent = `${formatOfferDetailKeys(key)}: `;
+    offerValue.textContent = value;
+    offerData.appendChild(offerKey);
+    offerData.appendChild(offerValue);
+    offerDetailsList.appendChild(offerData);
+  });
+  const checkoutLink = document.createElement('a');
+  checkoutLink.textContent = 'Checkout link';
+  const perpetual = searchParams.get('perp') === 'true' || undefined;
+  const promotionCode = searchParams.get('promo') || undefined;
+  const options = omitNullValues({
+    perpetual,
+    promotionCode,
+    wcsOsi: searchParams.get('osi'),
+    ...getCheckoutContext(searchParams, getConfig()),
+  });
+  const checkoutUrl = buildCheckoutButton(checkoutLink, options);
+  checkoutUrl.target = '_blank';
+  offerDetailsList.appendChild(checkoutUrl);
+  el.append(offerDetailsList);
+};
+
+export const handleSearch = async (e, el) => {
+  let searchParams = {};
+  function displaySearchError() {
+    const notValidUrl = document.createElement('h4');
+    notValidUrl.classList.add('not-valid-url');
+    notValidUrl.textContent = 'Not a valid offer link';
+    el.append(notValidUrl);
+  }
+  el.textContent = '';
+  const search = e.target.value;
+  if (search === '') return undefined;
+  try {
+    searchParams = new URL(search).searchParams;
+    const osi = searchParams.get('osi');
+    if (!osi) {
+      displaySearchError();
+      return undefined;
+    }
+    window.tacocat.wcs.resolveOfferSelector(osi).then(([offerDetails]) => {
+      decorateOfferDetails(el, offerDetails, searchParams);
+    });
+  } catch (error) {
+    displaySearchError();
+    return undefined;
+  }
+};
+
+export const decorateSearch = (el) => {
+  const search = createTag('input', { class: 'offer-search', placeholder: 'Enter offer URL to preview' });
+  const icon = createTag('div', { class: 'offer-search-icon' });
+  const searchWrapper = createTag('div', { class: 'offer-search-wrapper' }, [search, icon]);
+  const offerDetailsWrapper = createTag('div', { class: 'offer-details-wrapper' });
+  el.append(searchWrapper);
+  el.append(offerDetailsWrapper);
+  search.addEventListener('keyup', (e) => { handleSearch(e, offerDetailsWrapper); });
```

Comment:
nit: you can debounce the search request by 500ms or something like this to issue less search requests

Reply:
Great tip, added 500ms timeout ✅